src/BUGSPrimitives/distributions.jl

```diff
@@ -358,6 +358,10 @@ function dbeta(a, b)
     return Beta(a, b)
 end
 
+# `Distributions.MvNormalCanon` provides this parameterization, but it errors on ReverseDiff (others, including 
+# ForwardDiff, Mooncake, etc. seems to be fine). The reason for error seems to be that `MvNormalCanon` uses 
+# `PDMat` and `PDMat` uses `PDMats.quad` which does `chol_upper` (https://github.com/JuliaStats/PDMats.jl/blob/5e7d88ec271df4bd12accc16eb56e7a9e14043fb/src/chol.jl#L49-L69)
+# and `ReverseDiff` errors when trying to set lower triangular part of the upper triangular matrix.
```

Comment:
@sunxd3, can you try to write a rule for `chol_upper`? Please take a look at Bijectors since we likely have rules already there. 

cc @mhauru

Reply:
See, https://github.com/TuringLang/Bijectors.jl/blob/d69a7976f2f5a8b357e0451dff53e913436a8478/ext/BijectorsReverseDiffChainRulesExt.jl#L28-L42

Reply:
A MWE is added (https://github.com/TuringLang/JuliaBUGS.jl/pull/280#issuecomment-2708873934).

I am not entirely certain a rule for `PDMats.chol_upper` will resolve this (it didn't appear in the stacktrace).

The priority of this issue is not very high, I want to resolve it sometime soon, but might not at this instant.